test/evm/foundry/local/Arbitrum_L3_Adapter.t.sol

```diff
@@ -0,0 +1,141 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import { Test } from "forge-std/Test.sol";
+import { MockERC20 } from "forge-std/mocks/MockERC20.sol";
+import { ERC20, IERC20 } from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import { IERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import { IL1StandardBridge } from "@eth-optimism/contracts/L1/messaging/IL1StandardBridge.sol";
+import { Rerouter_Adapter } from "../../../../contracts/chain-adapters/Rerouter_Adapter.sol";
+import { Optimism_Adapter } from "../../../../contracts/chain-adapters/Optimism_Adapter.sol";
+import { WETH9Interface } from "../../../../contracts/external/interfaces/WETH9Interface.sol";
+import { ITokenMessenger } from "../../../../contracts/external/interfaces/CCTPInterfaces.sol";
+
+// We normally delegatecall these from the hub pool, which has receive(). In this test, we call the adapter
+// directly, so in order to withdraw Weth, we need to have receive().
+contract Mock_Rerouter_Adapter is Rerouter_Adapter {
+    constructor(address _l1Adapter, address _l2Target) Rerouter_Adapter(_l1Adapter, _l2Target) {}
+
+    receive() external payable {}
+}
+
+contract Token_ERC20 is ERC20 {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {}
+
+    function mint(address to, uint256 value) public virtual {
+        _mint(to, value);
+    }
+
+    function burn(address from, uint256 value) public virtual {
+        _burn(from, value);
+    }
+}
+
+contract MinimalWeth is Token_ERC20 {
+    constructor(string memory name, string memory symbol) Token_ERC20(name, symbol) {}
+
+    function withdraw(uint256 amount) public {
+        _burn(msg.sender, amount);
+        (bool success, ) = payable(msg.sender).call{ value: amount }("");
+        require(success);
+    }
+}
+
+contract CrossDomainMessenger {
+    event MessageSent(address indexed target);
+
+    function sendMessage(
+        address target,
+        bytes calldata,
+        uint32
+    ) external {
+        emit MessageSent(target);
+    }
+}
+
+contract StandardBridge {
```

Comment:
None of these test contracts are used in other tests? If so we should to try import them.

Reply:
Refactored https://github.com/across-protocol/contracts/pull/607/commits/6d8a6339a81368fc47271c89581408d5d8b94f1b